### ISSUE TITLE
[firebase_cloud_messaging] iOS Keychain usage vulnerability

### ISSUE THREAD

**Issue by VladimirKalinichenkoWeather** [FirebaseExtended/flutterfire#1966]
Throughout the firebase plugin, `kSecAttrAccessibleAlwaysThisDeviceOnly` accessibility option is used for storing data in KeyChain. Items added to the KeyChain with an insecure accessibility option may be exposed to other applications on JailBroken devices or attackers with physical access. 

Currently, there is no API to change the accessibility options used by Firebase. Insecure data storage is a secure compliance violation of our company's policies. We would like to be able to override default  Firebase keychain usage.
